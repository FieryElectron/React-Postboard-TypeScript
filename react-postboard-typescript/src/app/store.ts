import { configureStore, ThunkAction, Action } from '@reduxjs/toolkit';

import signInPageReducer from '../components/SignInPage/signInPageSlice';
import postBoardPageReducer from '../components/PostBoardPage/postBoardPageSlice';

export const store = configureStore({
  reducer: {
    signInPage: signInPageReducer,
    postBoardPage: postBoardPageReducer,
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
