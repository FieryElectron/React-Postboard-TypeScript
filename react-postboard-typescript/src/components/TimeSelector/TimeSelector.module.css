.timeSelectorTitle{
    margin: 5px;
    padding: 0px;
}

.timeSelectorContainer {
    display: flex;
    justify-content: center;
    align-items: center;
}

