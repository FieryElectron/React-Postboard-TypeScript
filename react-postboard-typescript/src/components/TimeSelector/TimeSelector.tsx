import styles from './TimeSelector.module.css';
import Button from '../Button/Button'

import moment from 'moment';
import { useEffect } from 'react';

import { useAppSelector, useAppDispatch } from '../../app/hooks';

import { useNavigate } from 'react-router-dom';

import {
    selectDateFrom,
    selectDateTo,
    setDateFrom,
    setDateTo,
    getPosts,
} from '../PostBoardPage/postBoardPageSlice';

const TimeSelector = () => {
    const dispatch = useAppDispatch();

    const fromDate = useAppSelector(selectDateFrom);
    const toDate = useAppSelector(selectDateTo);
    const navigate = useNavigate();

    useEffect(() => {
        const dateString = moment(new Date()).format('YYYY-MM-DD');
        dispatch(setDateFrom(dateString));
        dispatch(setDateTo(dateString));
    }, [dispatch]);


    const dateFromOnchange = (e: React.ChangeEvent<HTMLInputElement>) => {
        dispatch(setDateFrom(e.target.value));
    }

    const dateToOnchange = (e: React.ChangeEvent<HTMLInputElement>) => {
        dispatch(setDateTo(e.target.value));
    }

    const RefreshPosts = () => {
        dispatch(getPosts(navigate));
    }

    return (
        <div data-testid='timeSelectorContainer' className={styles.timeSelectorContainer}>
            <h3 data-testid='timeSelectorTitle1' className={styles.timeSelectorTitle}>From</h3>
            <input data-testid='timeSelectorDate1' id='dateFrom' type='date' className={styles.timeSelectorDate} onChange={dateFromOnchange} value={fromDate}></input>
            <h3 data-testid='timeSelectorTitle2' className={styles.timeSelectorTitle}>To</h3>
            <input data-testid='timeSelectorDate2' id='dateTo' type='date' className={styles.timeSelectorDate} onChange={dateToOnchange}  value={toDate}></input>
            <Button text='Refresh Posts' onClick={RefreshPosts} />
        </div>
    )
}

export default TimeSelector;