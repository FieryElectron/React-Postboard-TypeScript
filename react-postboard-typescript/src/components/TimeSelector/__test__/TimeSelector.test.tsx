import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { act } from 'react-dom/test-utils';
import '@testing-library/jest-dom/extend-expect';

import TimeSelector from '../TimeSelector';

import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import { store } from '../../../app/store';

let container: HTMLElement;

beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  document.body.removeChild(container);
  container.remove();
});

it('empty test', () => {

});

it('renders TimeSelector without crashing', () => {
  act(() => {
    render(<Provider store={store}><Router><TimeSelector /></Router></Provider>);
  });
});

it('renders TimeSelector class correctly', () => {
  act(() => {
    render(<Provider store={store}><Router><TimeSelector /></Router></Provider>);
  });

  const timeSelectorContainer = document.querySelector('[data-testid=timeSelectorContainer]')!;
  const timeSelectorTitle1 = document.querySelector('[data-testid=timeSelectorTitle1]')!;
  const timeSelectorDate1 = document.querySelector('[data-testid=timeSelectorDate1]')!;
  const timeSelectorTitle2 = document.querySelector('[data-testid=timeSelectorTitle2]')!;
  const timeSelectorDate2 = document.querySelector('[data-testid=timeSelectorDate2]')!;


  expect(timeSelectorContainer).toHaveAttribute('class', 'timeSelectorContainer');
  expect(timeSelectorTitle1).toHaveAttribute('class', 'timeSelectorTitle');
  expect(timeSelectorDate1).toHaveAttribute('class', 'timeSelectorDate');
  expect(timeSelectorTitle2).toHaveAttribute('class', 'timeSelectorTitle');
  expect(timeSelectorDate2).toHaveAttribute('class', 'timeSelectorDate');

});


 
it('matchs snapshot 1', ()=>{
    const tree = renderer.create(<Provider store={store}><Router><TimeSelector/></Router></Provider>).toJSON();
    expect(tree).toMatchSnapshot();
  });