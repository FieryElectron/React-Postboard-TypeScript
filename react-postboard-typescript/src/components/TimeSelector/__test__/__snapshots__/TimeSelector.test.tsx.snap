// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matchs snapshot 1 1`] = `
<div
  className="timeSelectorContainer"
  data-testid="timeSelectorContainer"
>
  <h3
    className="timeSelectorTitle"
    data-testid="timeSelectorTitle1"
  >
    From
  </h3>
  <input
    className="timeSelectorDate"
    data-testid="timeSelectorDate1"
    id="dateFrom"
    onChange={[Function]}
    type="date"
    value="2022-04-30"
  />
  <h3
    className="timeSelectorTitle"
    data-testid="timeSelectorTitle2"
  >
    To
  </h3>
  <input
    className="timeSelectorDate"
    data-testid="timeSelectorDate2"
    id="dateTo"
    onChange={[Function]}
    type="date"
    value="2022-04-30"
  />
  <button
    className="button"
    data-testid="button"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "#0071FE",
      }
    }
  >
    Refresh Posts
  </button>
</div>
`;
